pkg/lib_test.go

```diff
@@ -292,3 +318,463 @@ func TestGetNamespacesBatches(t *testing.T) {
+			TestNamespace4,
+			TestNamespace5,
+		},
+		Containers: []inventory.Container{
+			Container1,
+			Container2,
+			Container3,
+			Container4,
+			Container5,
+		},
+		Pods: []inventory.Pod{
+			Pod1,
+			Pod2,
+			Pod3,
+			Pod4,
+			Pod5,
+		},
+		Nodes: []inventory.Node{
+			Node1,
+			Node2,
+			Node3,
+		},
+	}
+)
+
+func Test_getBatchedInventoryReports(t *testing.T) {
+	type args struct {
+		reports   AccountRoutedReports
+		batchSize int
+	}
+	tests := []struct {
+		name string
+		args args
+		want BatchedReports
+	}{
+		{
+			name: "empty reports",
+			args: args{
+				reports:   AccountRoutedReports{},
+				batchSize: 10,
+			},
+			want: BatchedReports{},
+		},
+		{
+			name: "no batches configured (batch size 0)",
+			args: args{
+				reports: AccountRoutedReports{
+					"account1": TestReport,
+				},
+				batchSize: 0,
+			},
+			want: BatchedReports{
+				"account1": {
+					TestReport,
+				},
+			},
+		},
+		{
+			name: "single batch (namespace count < batch size)",
+			args: args{
+				reports: AccountRoutedReports{
+					"account1": TestReport,
+				},
+				batchSize: 10,
+			},
+			want: BatchedReports{
+				"account1": {
+					TestReport,
+				},
+			},
+		},
+		{
+			name: "single batch (namespace count == batch size)",
+			args: args{
+				reports: AccountRoutedReports{
+					"account1": TestReport,
+				},
+				batchSize: 5,
+			},
+			want: BatchedReports{
+				"account1": {
+					TestReport,
+				},
+			},
+		},
+		{
+			name: "multiple batches",
+			args: args{
+				reports: AccountRoutedReports{
+					"account1": TestReport,
+				},
+				batchSize: 1,
+			},
+			want: BatchedReports{
+				"account1": {
+					{
+						ClusterName: "cluster1",
+						Namespaces:  []inventory.Namespace{TestNamespace1},
+						Containers:  []inventory.Container{Container1},
+						Pods:        []inventory.Pod{Pod1},
+						Nodes:       []inventory.Node{Node1},
+					},
+					{
+						ClusterName: "cluster1",
+						Namespaces:  []inventory.Namespace{TestNamespace2},
+						Containers:  []inventory.Container{Container2},
+						Pods:        []inventory.Pod{Pod2},
+						Nodes:       []inventory.Node{Node2},
+					},
+					{
+						ClusterName: "cluster1",
+						Namespaces:  []inventory.Namespace{TestNamespace3},
+						Containers:  []inventory.Container{Container3},
+						Pods:        []inventory.Pod{Pod3},
+						Nodes:       []inventory.Node{Node2},
+					},
+					{
+						ClusterName: "cluster1",
+						Namespaces:  []inventory.Namespace{TestNamespace4},
+						Containers:  []inventory.Container{Container4},
+						Pods:        []inventory.Pod{Pod4},
+						Nodes:       []inventory.Node{Node3},
+					},
+					{
+						ClusterName: "cluster1",
+						Namespaces:  []inventory.Namespace{TestNamespace5},
+						Containers:  []inventory.Container{Container5},
+						Pods:        []inventory.Pod{Pod5},
+						Nodes:       []inventory.Node{Node3},
+					},
+				},
+			},
+		},
+		{
+			name: "multiple batches (individual namespaces)",
```

Comment:
This looks the same as the previous test case?

Reply:
I does indeed appear to be the same, must have fat fingered it I'll delete it